x-pack/plugins/fleet/server/routes/epm/handlers.ts

```diff
@@ -87,12 +87,16 @@ export const getListHandler: FleetRequestHandler<
 > = async (context, request, response) => {
   try {
     const savedObjectsClient = (await context.fleet).internalSoClient;
+    const { limitedToPackages } = await context.fleet;
     const res = await getPackages({
       savedObjectsClient,
       ...request.query,
     });
+
+    const items = filterAllowedPackages(res, 'id', limitedToPackages);
```

Comment:
so up until this change, we have been failing the API if the data contained more than what they are allowed to see. This change is now filtering out packages instead of failing the request, so that behaviour differs from that approach. I would like to see if we can stay with that model before returning data filtered out. 

⚠️ The bigger issue I see is that this response will be "weird" because of the pagination query params - ex. user requested 10 per page and lets say there are 100 total items. They would potentially get back a response that indicates the total is like `100` but the content of this one page could have less than the `perPage` value in the query params. That's not a good solution. 

I just checked the EPM APIs from fleet and it looks like there is now an API that we can use to retrieve only the Endpoint package - `/api/fleet/epm/packages/endpoint`. I think [we should change the implementation on the security solution](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/services/policies/ingest.ts#L136-L148) side to use this API instead and update that API (not this one) to support package privileges . 


**Some History:**

If I remember correctly, we (in security solution) only need to get the version number of the Endpoint package in order to be able to create a URL jump to the integration Package detail page (ex. `http://localhost:5601/app/integrations/detail/endpoint-8.6.0/overview`). The only available way to do that at the time (FYI: FLeet was not even released back then, so everything was very new) was via the List EPM api using a filter of `category=security` and then finding the Endpoint package in that list. We do this because:

1. The API here does not have a way to filter by `package.name`
2. The integrations details page does not support using only the `package.name` in the URL

But recognizing that we now seem to have a better api, lets update our code to use that instead.

Reply:
In light of this context, switching to `/api/fleet/epm/packages/endpoint` definitely sounds better.

A bit moot since I agree we should just switch APIs but the reason I had to take a filter approach instead of 403 immediately on unauthorized package is due to the fact that this API would also 403 since, as you mentioned, we can't filter by package name.